### PR TITLE
Goodbye jshint

### PR DESCRIPTION
**Pull request by yiyangli-okta**
Resolves: OKTA-121686

https://travis-ci.org/okta/okta-signin-widget/builds/219002269#L2780

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
     "grunt-contrib-copy": "0.8.0",
     "grunt-contrib-jasmine": "0.9.2",
     "grunt-contrib-jshint": "0.11.2",
+    "grunt-eslint": "^17.3.1",
```
Comment (haishengwu-okta) [inline package.json]: any chance to go with version 19.0.0?

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  },
+  "parserOptions": {
+    "ecmaVersion": 3
+  },
+  "globals": {
+    "spyOnEvent": false,
+    "JSON": true
+  },
+  rules: {
+    "camelcase": 1,
```
Comment (haishengwu-okta) [inline .eslintrc]: shall be fine to set to 2 (error) rather than 1 (warning). if we want grunt to keep running regardless error, `--force` is the trick.

Comment (yiyangli-okta) [inline .eslintrc]: oh, I copied the `.eslintrc` from okta-core, let me try.

**Comment by yiyangli-okta** [comment]
Merge request @rchild-okta, thanks.

File: src/EnrollDuoController.js [hunk 3]
```diff
@@ -10,15 +10,15 @@
...
+/* eslint camelcase: 0*/
 define([
   'okta',
   'duo',
   'vendor/lib/q',
   'util/FormController',
   'views/enroll-factors/Footer'
 ],
-function (Okta, Duo, Q, FormController, Footer) {
+function (Okta, Duo, q, FormController, Footer) {
```
Comment (rchild-okta) [inline src/EnrollDuoController.js]: Can we disable this rule (i.e. with jshint it was the newcap flag)? Ideally we would reference Q the same way as in the q docs.

**Comment by yiyangli-okta** [comment]
Thanks @rchild-okta , I will let you know once I complete them. Sorry for the delay.

File: .eslintrc [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    "complexity": [2, 6],
+    "curly": 2,
+    "dot-notation": 0,
+    "guard-for-in": 2,
+    "indent": [2, 2, { "VariableDeclarator": 2 } ],
+    "max-depth": [2, 2],
+    "max-len": [2, 120],
+    "max-params": [2, 10],
+    "max-statements": [2, 20],
+    "new-cap": [2, { properties: false, capIsNewExceptions: ["Q"] }],
```
Comment (yiyangli-okta) [inline .eslintrc]: `Q` is an exception for capIsNew

File: Gruntfile.js [hunk 5]
```diff
@@ -20,12 +20,24 @@ module.exports = function (grunt) {
...
+      JS_LINT_FILES         = [
+        'Gruntfile.js',
+        'src/*.js',
+        'src/**/*.js',
+        '!src/vendor/*.js',
+        '!src/util/countryCallingCodes.js',
+        'buildtools/**/*.js',
+        '!buildtools/r.js',
+        'test/unit/helpers/**/*.js',
+        'test/**/**/*.js',
```
Comment (yiyangli-okta) [inline Gruntfile.js]: I didn't realize that js test files are changed from `test/**/*.js` to `test/unit/**/*.js`, now it's good to review and merge.

File: test/e2e/.eslintrc.json [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{
+  "env": {
+    "browser": true,
+    "jasmine": true,
+    "node": true,
+    "protractor": true,
+    "es6": true
```
Comment (yiyangli-okta) [inline test/e2e/.eslintrc.json]: for e2e that uses `es6` syntax.

**Comment by yiyangli-okta** [comment]
ready to merge

**Pull request merged**